BookingsApi/BookingsApi.DAL/Services/HearingAllocationService.cs

```diff
@@ -38,40 +39,124 @@ public class HearingAllocationService : IHearingAllocationService
             _configuration = configuration.Value;
             _logger = logger;
         }
-        
-        public async Task<JusticeUser> AllocateAutomatically(Guid hearingId)
+
+        /// <summary>
+        /// Allocate a list of hearings to a passed CSO.
+        /// </summary>
+        /// <param name="postRequestHearings">List of hearings to be allocated</param>
+        /// <param name="postRequestCsoId">CSO JusticeUser Id</param>
+        /// <exception cref="DomainRuleException"></exception>
+        public async Task<List<VideoHearing>> AllocateHearingsToCso(List<Guid> postRequestHearings, Guid postRequestCsoId)
```

Comment:
I imagine the logic for automatic and manual have some difference, why not create a shared private method for the common parts and make the allocate automatically and allocate manually two separate methods. reduce the responsibility of this method so it's easier to follow and maintain

Reply:
not much difference between automatic and manual. If we do as you suggested we end up in duplicate code.

Reply:
I agree with Shaed, a separate method for manual and automatic allocation would be cleaner and easier to understand here.

The only code common to both is where it adds the allocation to the database, which you have already refactored into a separate method (AllocateHearingToCso). So they could both call that

Reply:
I have created 2 separate methods.